File: app/src/main/java/com/osrsd/command/PrintHealthBars.java
package com.osrsd.command;

import lombok.AllArgsConstructor;

import java.util.concurrent.CountDownLatch;

@AllArgsConstructor
public class PrintHealthBars implements Runnable {

    private final CountDownLatch latch;

    @Override
    public void run() {
        long start = System.currentTimeMillis();
//        Library.cachePrintable(HealthbarDefinition.class, new HealthBarLoader().load().getDefinitions());
//        App.prompt(PrintHealthBars.class, start);
        latch.countDown();
    }

}

File: app/src/main/java/com/osrsd/command/PrintHealthBars.kt
package com.osrsd.command

import com.osrsd.cache.Library
import com.osrsd.cache.loader.EnumLoader
import com.osrsd.cache.loader.HealthBarLoader
import com.osrsd.cache.loader.ItemLoader
import com.osrsd.cache.loader.NpcLoader
import com.osrsd.spring.App
import com.osrsd.spring.domain.EnumDefinition
import com.osrsd.spring.domain.HealthBarDefinition
import com.osrsd.spring.domain.ItemDefinition
import com.osrsd.spring.domain.NpcDefinition
import java.util.concurrent.CountDownLatch

class PrintHealthBars(val latch: CountDownLatch) : Runnable {

    override fun run() {
        val start = System.currentTimeMillis()
        Library.store(HealthBarDefinition::class.java, HealthBarLoader().load().definitions)
        App.prompt(PrintHealthBars::class.java, start)
        latch.countDown()
    }

}